src/processor.ts

```diff
@@ -162,12 +163,53 @@ export class Processor {
     return references;
   }
 
+  makeSortKey(key: string, reference: ProcReference): string | undefined {
+    switch (key) {
+      case "author": {
+        const authors = reference.author;
+        if (authors !== undefined) {
+          // FIX this is wrong
+          const formattedAuthors = authors.map((author) => author.name);
+          return formattedAuthors.toString();
+        }
+        break;
+      }
+      case "year": {
+        const year = reference.issued.getFullYear().toString();
+        return year || "";
+      }
+      // TODO complete for all key options
+      case "title":
+        // REVIEW make sure this actually works
+        return reference.title.toString();
+    }
+  }
+
+  sortReferences(sortConfig: SortConfig[]): ProcReference[] {
+    // REVIEW will this actually work?
```

Comment:
I don't think so.

Think I need to check the length, and then based on that the subsequent code.

For example, if length is two then `... || ...`.